src/views/NewFlow.vue

```diff
@@ -0,0 +1,95 @@
+<template>
+  <div class="new-contents">
+    <div v-if="activeFlow" class="row">
+      <div class="col-sm-8 offset-sm-2">
+        <div class="card">
+          <div class="card-body">
+            <div v-if="flowError" class="alert alert-danger" role="alert">
+              {{flowError | trans}}
+            </div>
+            <flow-editor :flow="activeFlow" flow-header="flow-builder.create-flow" :sidebar=false />
+
+            <div class="float-right">
+              <router-link :to="route('trees.editTree', {treeId: activeFlow.uuid, component: 'interaction-designer', mode: 'edit'})"
+                class="btn btn-primary"
+                event=""
+                @click.native.prevent="handlePersistFlow(route('trees.editTree', {treeId: activeFlow.uuid, component: 'interaction-designer', mode: 'edit'}))">
+                {{'flow-builder.save-and-continue' | trans}}
+              </router-link>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+
+import FlowEditor from '@/components/interaction-designer/flow-editors/FlowEditor.vue'
+import lang from '@/lib/filters/lang'
+import Routes from '@/lib/mixins/Routes'
+import { Component, Prop } from 'vue-property-decorator'
+import Vue from 'vue'
+import {Getter, Mutation, namespace} from 'vuex-class'
+import { forEach, isEmpty } from 'lodash'
+import {store} from '@/store'
+const flowVuexNamespace = namespace('flow')
+import {IFlow, IContext} from '@floip/flow-runner'
+
+@Component<any>(
+  {
+    components: {
+      FlowEditor,
+    },
+    mixins: [lang, Routes],
+    async mounted() {
+        await this.flow_addBlankFlow()
```

Comment:
@bzabos Found a bug with this today that I've been wrestling with but I think I've not decided to leave fixing for now. Basically, when you hit this page a blank flow is added to the container. When you "Save and continue" we hit the backend dummy route for persisting and move to the interaction designer. The issue is, if you came to this page from the link on Home.vue and then instead of submitting you go back to the homepage with the back button (which vue-router handles) you will see the empty flow in the list there - even though you never saved it.

I think a fix for this is a bit tricky as all of the flow-editing components edit the currently active flow which comes from the active flow container. It would be a big-ish refactor to let these components edit this or alternatively edit a temporary flow that is not yet in the container. This might also result in a pretty ugly component. 

The alternative is to set some sort of isPersisted/isCreated flag on the flows when we persist and then not render un persisted/created flows on the home page. Then we can find some point in the flow lifecycle to clear out unpersisted flows. This feels a bit hacky though.

As I understand it, we may in future have a draft state that we can use here and do something like option 1 in a more though out way. This coupled with the fact that this Home component won't actually ever be used by voto5 (the flows index will end up coming from php there and any navigation to this page will pull only the actually persisted flows) means I don't think this is a big enough issue to invest time in fixing now. Let me know if you disagree though.

Reply:
Nice! To your conclusion: I agree with you, no need to fix for now! Can we create a JIRA to revisit this in `2.0`-ish?

Overall, I'm partial to the `draft` concept where we would edit a local flow, then hitting _Continue_ would then add the flow to the state. 

Begs the question of whether or not we should update our approach to vuex actions API to be more of a functional toolset like lodash, eg: `setFlowName("My First Flow", {flow})` which would then modify the _reference_ (rather than performing lookup from state) while simultaneously continuing to provide an encapsulation of any logic needed behind the scenes.